views/components/Maps/Maps.tsx

```diff
@@ -53,63 +56,107 @@ export const printObjectInfo = (obj: any): any[] => {
  * @param obj - object with fields latitude and longitude
  * @returns an array containing [latitude, longitude]
  */
-export const convertToLatLng = (obj: any): LatLngExpression => {
-  return latLng(obj.latitude, obj.longitude);
-};
+export const convertToLatLng = (obj: any): L.LatLngExpression => {
+    return L.latLng(obj.latitude, obj.longitude)
+}
 
 export default class Maps extends React.Component<IMapsProps, IMapsState> {
-  renderShips = () => {
-    return this.props.aisShips.map((ship, index) => {
-      return (
-        <Circle
-          key={index}
-          center={[ship.latitude, ship.longitude]}
-          radius={50}
-          pathOptions={{ color: 'red' }}
-        >
-          <Popup>{printObjectInfo(ship)}</Popup>
-        </Circle>
-      );
-    });
-  };
-
-  render() {
-    return (
-      <MapContainer
-        center={convertToLatLng(this.props.gpsLocation)}
-        zoom={13}
-        scrollWheelZoom={true}
-        style={{ height: '100vh', width: '100wh' }}
-      >
-        <TileLayer
-          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
-          url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
-        />
-        <LayersControl position='bottomleft'>
-          <LayersControl.Overlay name='AIS Ships' checked>
-            <LayerGroup>{this.renderShips()}</LayerGroup>
-          </LayersControl.Overlay>
-          <LayersControl.Overlay name='Local Path' checked>
-            <Polyline
-              pathOptions={{ color: 'red' }}
-              positions={[this.props.localPath]}
-            />
-          </LayersControl.Overlay>
-          <LayersControl.Overlay name='Global Path' checked>
-            <Polyline
-              pathOptions={{ color: 'black', opacity: 0.25 }}
-              positions={[this.props.globalPath]}
-            />
-          </LayersControl.Overlay>
-        </LayersControl>
-        <Marker position={convertToLatLng(this.props.gpsLocation)}>
-          <Popup>{printObjectInfo(this.props.gpsLocation)}</Popup>
-        </Marker>
-        <Polyline
-          pathOptions={{ color: 'black' }}
-          positions={this.props.gpsPath}
-        />
-      </MapContainer>
-    );
-  }
+    readonly state: IMapsState = {
+        map: null,
+    };
+
+    setMapRef = (map: L.Map) => {
+        this.setState((state) => ({ ...state, map: map }));
+    }
+
+    rotatePoint = (point: L.LatLngExpression, angle: number, axis: L.LatLngExpression) => {
```

Comment:
Let's add a method description for this. It can be brief. Please take a look at the methods above, e.g. `convertToLatLng` or `printObjectInfo` for examples.